src/include/catalog.h

```diff
@@ -0,0 +1,24 @@
+#include <vector>
+#include "table.h"
+#include <string>
+
+namespace emerald {
+    class Catalog {
+        private:
+          std::vector<Table*> TableList;
+        public:
+          Catalog();
+          void AddTable(DbFile* file, std::string name, std::string pkeyField);
+          void AddTable(DbFile* file, std::string name);
+          void AddTable(DbFile* file);
+          int get_table_id(std::string name);
+          TupleDesc* GetTupleDesc(int tableid);
+          std::vector<Table*> get_tables();
```

Comment:
I think this function isn't needed? But I think it can be useful so good job on adding it!